lib/web_ui/lib/initialization.dart

```diff
@@ -23,94 +23,66 @@
-/// * https://github.com/flutter/flutter/blob/61fb2de52c7bdac19b7f2f74eaf3f11237e1e91d/packages/flutter_tools/lib/src/isolated/resident_web_runner.dart#L460-L485
-///
-/// This function first calls [engine.initializeEngineServices] so the engine
-/// can prepare the js-interop layer that is used by web apps (instead of the
-/// old `ui.webOnlyFoo` methods/getters).
-///
-/// It then creates a JsObject that is passed to the [didCreateEngineInitializer]
-/// JS callback, to delegate bootstrapping the app to the programmer.
-///
-/// If said callback is not defined, this assumes that the Flutter Web app is
-/// initializing "automatically", as was normal before this feature was
-/// introduced. This will immediately run the initEngine and runApp methods
-/// (via [engine.AppBootstrap.now]).
-///
-/// This is the only bit of `dart:ui` that should be directly called by Flutter
-/// web apps. Everything else should go through the JS-interop layer created in
-/// `engine.warmup`.
-///
-/// This method should NOT trigger the download of any additional resources
-/// (except when the app is in "autoStart" mode).
+// TODO(mdebbar): Deprecate this and remove it.
+// https://github.com/flutter/flutter/issues/127395
 Future<void> webOnlyWarmupEngine({
   Function? registerPlugins,
   Function? runApp,
-}) async {
-  // Create the object that knows how to bootstrap an app from JS and Dart.
-  final engine.AppBootstrap bootstrap = engine.AppBootstrap(
-    initializeEngine: ([engine.JsFlutterConfiguration? configuration]) async {
-      await engine.initializeEngineServices(jsConfiguration: configuration);
-    }, runApp: () async {
-      if (registerPlugins != null) {
-        registerPlugins();
-      }
-      await engine.initializeEngineUi();
-      if (runApp != null) {
-        runApp();
-      }
-    },
+}) {
+  assert(() {
+    engine.printWarning(
+      'The webOnlyWarmupEngine API is deprecated and will be removed in a '
+      'future release. Please use `warmupEngine` from `dart:ui_web` instead.',
+    );
+    return true;
+  }());
+  return ui_web.warmupEngine(
+    registerPlugins: registerPlugins,
+    runApp: runApp,
   );
+}
 
-  final engine.FlutterLoader? loader = engine.flutter?.loader;
-  if (loader == null || loader.isAutoStart) {
-    // The user does not want control of the app, bootstrap immediately.
-    engine.domWindow.console.debug('Flutter Web Bootstrap: Auto.');
-    await bootstrap.autoStart();
-  } else {
-    // Yield control of the bootstrap procedure to the user.
-    engine.domWindow.console.debug('Flutter Web Bootstrap: Programmatic.');
-    loader.didCreateEngineInitializer(bootstrap.prepareEngineInitializer());
-  }
+// TODO(mdebbar): Deprecate this and remove it.
+// https://github.com/flutter/flutter/issues/127395
+bool get debugEmulateFlutterTesterEnvironment {
```

Comment:
Is this something that is needed for end users, or is this only for our testing? If it's only for our engine testing, it probably should live within `engine` or similar, rather than be exposed to `web_ui`?

Reply:
(Never mind, this is used all over the bootstrap and the framework!)

Reply:
Alternatively, this could be hidden behind JS-interop, if we don't want it to show up in the public APIs.

Reply:
`debugEmulateFlutterTesterEnvironment` has effect on several things including the usage of Ahem font. If users want their tests to produce real text (e.g. for screenshots), they have to disable `debugEmulateFlutterTesterEnvironment`.